src/pixelator/pixeldataset/datastores.py

```diff
@@ -0,0 +1,507 @@
+
+        :param dataframe: The dataframe to write.
+        :param key: The key to write the dataframe to.
+        :param partitioning: The (optional) partitioning to use when
+                             writing the dataframe.
+        """
+        ...
+
+    def read_edgelist(self) -> pd.DataFrame:
+        """Read an edgelist from the pixel data store.
+
+        :return: The edgelist.
+        :rtype: pd.DataFrame
+        """
+        ...
+
+    def read_edgelist_lazy(self) -> pd.DataFrame:
+        """Read a lazy edgelist from the pixel data store.
+
+        :return: The lazy edgelist.
+        :rtype: pd.DataFrame
+        """
+        ...
+
+    def read_polarization(self) -> Optional[pd.DataFrame]:
+        """Read the polarization data from the pixel data store.
+
+        :return: The polarization data, or None if it doesn't exist.
+        :rtype: Optional[pd.DataFrame]
+        """
+        ...
+
+    def read_colocalization(self) -> Optional[pd.DataFrame]:
+        """Read the colocalization data from the pixel data store.
+
+        :return: The colocalization data, or None if it doesn't exist.
+        :rtype: Optional[pd.DataFrame]
+        """
+        ...
+
+    def write_edgelist(
+        self, edgelist: pd.DataFrame, partitioning: Optional[list[str]] = None
+    ) -> None:
+        """Write the given edgelist to the pixel data store.
+
+        If a partitioning is provided, the edgelist will be partitioned accordingly.
+        This may not be supported by all data stores. If not supported, this
+        option will be ignored.
+
+        :param edgelist: The edgelist to write.
+        :param partitioning: The (optional) partitioning to use when
+                             writing the dataframe.
+        """
+        ...
+
+    def write_polarization(self, polarization: pd.DataFrame) -> None:
+        """Write the given polarization data to the pixel data store.
+
+        :param polarization: The polarization data to write.
+        """
+        ...
+
+    def write_colocalization(self, colocalization: pd.DataFrame) -> None:
+        """Write the given colocalization data to the pixel data store.
+
+        :param colocalization: The colocalization data to write.
+        """
+        ...
+
+    def save(self, dataset: PixelDataset) -> None:
+        """Save the given PixelDataset to the pixel data store.
+
+        :param dataset: The PixelDataset to save.
+        """
+        ...
+
+
+class PixelDataStoreError(Exception):
+    """Base class for all PixelDataStore related exceptions."""
+
+    pass
+
+
+class FileFormatNotRecognizedError(PixelDataStoreError):
+    """Raised when the file format of a .pxl file is not recognized."""
+
+    pass
+
+
+class EdgelistNotFoundError(PixelDataStoreError):
+    """Raised when the edgelist is not found in a .pxl file."""
+
+    pass
+
+
+class ZipBasedPixelFile(PixelDataStore):
+    """Superclass for all zip-based pixel data stores."""
+
+    FILE_FORMAT_VERSION: int = 1
```

Comment:
Right now I think it is ok to keep it like this, and I suggest only keeping a major file format version around since an increment should indicate breaking functionality. So far everything we have will be backwards compatible (with the exception that csv-based edgelist cannot be lazy loaded, but they are not really around in the wild anyway) so we should be fine for staying at version 1.

Reply:
Ok